### PR TITLE
Feat/hycu add download license

### PR DESCRIPTION
**Pull request by tibs245**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/hycu`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Feat #MANAGER-14499
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

### CODE REVIEW

File: packages/manager/apps/hycu/src/constants.ts [hunk 1]
```diff
@@ -7,3 +7,6 @@ export const packTypeLabel = {
   'hycu-cloud-vm-pack-250': '250 VMs',
   'hycu-cloud-vm-pack-500': '500 VMs',
 } as const;
+
+export const PREFIX_LICENSE_FILE_NAME = 'license-hycu-' as const;
+export const SUFFIX_LICENSE_FILE_NAME = '.dat' as const;
```
Comment (pauldkn) [inline packages/manager/apps/hycu/src/constants.ts:12]: Not sure it's useful here to specify `as const` like in an "object vs enum" case

**Review by pauldkn — APPROVED** [review]

File: packages/manager/apps/hycu/src/constants.ts (hunk 1 above)
Comment (darsene) [reply packages/manager/apps/hycu/src/constants.ts:12]: Why not consider using a kind of name template instead ?
```suggestion
export const LICENSE_FILE_NAME_TEMPLATE = 'license-hycu-{filename}.dat' as const;
```

File: packages/manager/apps/hycu/src/hooks/api/license.ts [hunk 2]
```diff
@@ -90,3 +98,29 @@ export const useRegenerateLicenseHYCUMutation = (
...
+  > = {},
+) => {
+  const { onSuccess, ...restOptions } = options;
+
+  return useMutation({
+    mutationKey: getDownloadLicenseHycuMutationKey(),
+    mutationFn: (params) => getLicenseHycuDownloadService(params),
+    onSuccess: (data, variables, context) => {
+      downloadTextAsFile(
+        `${PREFIX_LICENSE_FILE_NAME}${variables.serviceName}${SUFFIX_LICENSE_FILE_NAME}`,
```
Comment (darsene) [inline packages/manager/apps/hycu/src/hooks/api/license.ts:118]:
```suggestion
        LICENSE_FILE_NAME_TEMPLATE.replace('{filemame}',variables.serviceName),
```

File: packages/manager/apps/hycu/src/utils/downloadTextAsFile.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+/* c8 ignore start */
```
Comment (darsene) [inline packages/manager/apps/hycu/src/utils/downloadTextAsFile.ts:1]: Should not be excluded from the coverage report IMO.
By the way, it's better to use c8 configuration file instead

File: packages/manager/apps/hycu/src/utils/iam.constants.ts [hunk 4]
```diff
@@ -1,4 +1,5 @@
 export const IAM_ACTIONS = {
+  licenseHycuApiOvhGet: 'licenseHycu:apiovh:license/get',
```
Comment (darsene) [inline packages/manager/apps/hycu/src/utils/iam.constants.ts:2]: Can `get` be finally managed with IAM ?

Comment (tibs245) [reply packages/manager/apps/hycu/src/utils/iam.constants.ts:2]: It's the rule for download license. I thinked It's relevant for this case

File: packages/manager/apps/hycu/public/translations/hycu/dashboard/Messages_fr_FR.json [hunk 5]
```diff
@@ -24,8 +24,9 @@
   "hycu_dashboard_wait_for_activation": "En attente d'activation",
   "hycu_dashboard_license_activate_heading": "Activer la licence",
   "hycu_dashboard_license_activate_description": "Veuillez transférer votre fichier de demande de licence. Ce fichier est à télécharger depuis votre controller HYCU une fois votre installation réalisée",
-  "hycu_dashboard_license_more_information_link": "(plus de détails ici)",
+  "hycu_dashboard_license_more_information_link": "(plus de détails ici).",
   "hycu_dashboard_activation_license_success_message": "La demande d'activation de votre licence a bien été prise en compte. Votre fichier de licence sera bientôt disponible sur cette interface et vous sera également envoyé par e-mail.",
+  "hycu_dashboard_regenerate_success_message": "La demande d'activation de votre licence a bien été prise en compte. Votre fichier de licence sera bientôt disponible sur cette interface et vous sera également envoyé par e-mail.",
```
Comment (pauldkn) [inline packages/manager/apps/hycu/public/translations/hycu/dashboard/Messages_fr_FR.json:29]: Duplicate translation

**Review by pauldkn — APPROVED** [review]

File: packages/manager/apps/hycu/public/translations/hycu/dashboard/Messages_fr_FR.json [hunk 6]
```diff
@@ -34,15 +35,17 @@
   "hycu_dashboard_regenerate_upload_confirm": "Regénérer",
   "hycu_dashboard_upload_cancel": "Annuler",
   "hycu_dashboard_upload_confirm": "Activer",
-  "hycu_dashboard_upload_valid_file": "Merci de sélectionner un fichier valide",
+  "hycu_dashboard_upload_file_too_big": "La taille du fichier ne peut pas excéder 1Mo",
+  "hycu_dashboard_upload_file_bad_type": "Le fichier fourni n'est pas un fichier .req",
```
Comment (darsene) [inline packages/manager/apps/hycu/public/translations/hycu/dashboard/Messages_fr_FR.json:39]: To easy any change, can you use variable instead (no further translation needed)
```suggestion
  "hycu_dashboard_upload_file_bad_type": "Le fichier fourni n'est pas un fichier {{extension}}",
```

File: packages/manager/apps/hycu/public/translations/hycu/dashboard/Messages_fr_FR.json [hunk 7]
```diff
@@ -34,15 +35,17 @@
...
   "hycu_dashboard_upload_license_required": "Veuillez sélectionner une licence",
   "hycu_dashboard_drag_and_drop_attachment": "Glissez-Déposer une pièce jointe",
-  "hycu_dashboard_accepted_formats": "Format accepté: .dat",
+  "hycu_dashboard_accepted_formats": "Format accepté: .req",
```
Comment (darsene) [inline packages/manager/apps/hycu/public/translations/hycu/dashboard/Messages_fr_FR.json:42]:
```suggestion
  "hycu_dashboard_accepted_formats": "Format accepté: {{extension}}",
```

File: packages/manager/apps/hycu/src/components/FileUpload/FileUpload.component.tsx [hunk 8]
```diff
@@ -31,14 +31,24 @@ export const FileInputField = ({
     if (file.size > MAX_FILE_SIZE) {
       setFileName('');
       control.setError(name, {
-        message: t('hycu_dashboard_upload_valid_file'),
+        message: t('hycu_dashboard_upload_file_too_big'),
+      });
+      return;
+    }
+    if (!file.name.endsWith('.req')) {
```
Comment (darsene) [inline packages/manager/apps/hycu/src/components/FileUpload/FileUpload.component.tsx:38]: To set in a constant to use also in translation key
```suggestion
    if (!file.name.endsWith(LICENSE_FILE_EXT)) {
```

File: packages/manager/apps/hycu/src/constants.ts [hunk 9]
```diff
@@ -7,3 +7,5 @@ export const packTypeLabel = {
   'hycu-cloud-vm-pack-250': '250 VMs',
   'hycu-cloud-vm-pack-500': '500 VMs',
 } as const;
+
+export const LICENSE_FILE_NAME_TEMPLATE = 'license-hycu-{serviceName}.dat';
```
Comment (darsene) [inline packages/manager/apps/hycu/src/constants.ts:11]:
```suggestion
export const LICENSE_FILE_NAME_TEMPLATE = 'license-hycu-{serviceName}.${LICENSE_FILE_EXT}';
```

File: packages/manager/apps/hycu/src/components/FileUpload/FileUpload.component.tsx [hunk 10]
```diff
@@ -31,14 +31,24 @@ export const FileInputField = ({
...
+      setFileName('');
+      control.setError(name, {
+        message: t('hycu_dashboard_upload_file_bad_type'),
```
Comment (darsene) [inline packages/manager/apps/hycu/src/components/FileUpload/FileUpload.component.tsx:41]:
```suggestion
        message: t('hycu_dashboard_upload_file_bad_type', {extension : LICENSE_FILE_EXT}),
```

**Review by darsene — CHANGES_REQUESTED** [review]

File: packages/manager/apps/hycu/public/translations/hycu/dashboard/Messages_fr_FR.json (hunk 5 above)
Comment (darsene) [reply packages/manager/apps/hycu/public/translations/hycu/dashboard/Messages_fr_FR.json:29]: @tibs245 Can you use the same key for both use cases

**Review by darsene — APPROVED** [review]

**Pull request merged**